packages/terra-form-validation/docs/ValidationFunctions.md

```diff
@@ -0,0 +1,68 @@
+# Validation Functions
+
+*react-final-form* gives you the ability to create your own validation functions. The key for react-final-form is that a validation function returns a blank message for when an input is invalid, and returns undefined when an input does not have any errors.
+
+Example validation functions
+
+```js
+const required = value => (value ? undefined : 'Required')
+
+const minLength = value => {
+  const MIN_LENGTH = 3;
+
+  if (value === undefined || value === null) {
+    return 'Empty Strings are not allowed'
+  }
+
+  if (value.length < MIN_LENGTH) {
+    return `String must be at least ${MIN_LENGTH} characters`;
+  }
+
+  return '';
+}
+```
+
+## Asynchronous Validations
+
+There maybe times you need to do remote validations such as for Unique Username checks. This involved awaiting for a Promise to resolve, and passing it's value off to the validation function.
+
+Example Asynchronous Validation
+
+```jsx
+const validateUniqueUser = async (name) => {
+  let response = new Promise((resolve, reject) => {
+    if (name !== 'Unique User') {
+      return resolve('');
+    } else {
+      return resolve('Name is Unavailable');
+    }
+  });
+
+  await response;
+  return response;
+}
+
+```
+
+*Note:* The promise can be any kind of promise, including calls with axios and other libraries.
+
+# Parsing
+
+Form inputs can be parsed to ensure input values follow specific formats. This could be useful when you want to restrict numeric inputs to only allow users to enter in digits, and force formating of phone numbers.
+
+```jsx
```

Comment:
Given the example validations in this file, how would one add this to their terra-form components to ensure these validations are correct? OnChange/OnSubmit? Is there a recommendation on where a validation like these examples would best be used?

Reply:
They wouldn't add the validation functions to the terra components. They would add them to the Field provided by react-final-form. Consumers just have to make sure that onChange and value are populated correctly for all of their inputs that are nested inside a react-final-form Field.

Reply:
I did however notice the lack of documentation on providing onChange and value for Terra components, so I added that to the documentation. I'll see if there are other places that it needs to be added to.